examples/examples/thread-info.rs

```diff
@@ -0,0 +1,43 @@
+#![deny(rust_2018_idioms)]
+/// This is a example showing how thread info can be enabled
+/// to be attached with events
+///
+/// You can run this example by running the following command in a terminal
+///
+/// ```
+/// cargo run --example thread-info
+/// ```
+use std::thread;
+use std::time::Duration;
+use tracing::info;
+
+fn main() {
+    tracing_subscriber::fmt()
+        .with_max_level(tracing::Level::DEBUG)
+        .with_thread_ids(true)
+        .with_thread_names(true)
```

Comment:
It could be good to add comments on these lines explaining what they're doing, since it's the main thing the example is intended to demonstrate.

Reply:
Also, take it or leave it: might be nice if the example had CLI arguments to control thread IDs and thread names separately, so that users can demo all the possible combinations? But, adding argument-parsing would add a bunch of code not directly related to the things we're trying to show in this example, so I am not sure whether or not this is a good idea. What do you think?